src/redux/actions/concepts/dictionaryConcepts.js

```diff
@@ -172,6 +165,52 @@ export const filterByClass = (
   dispatch(fetchDictionaryConcepts(conceptType, conceptOwner, conceptName, query));
 };
 
+export const queryPossibleAnswers = query => async (dispatch) => {
+  const url = `concepts/?q=${query}&limit=0`;
+  try {
+    const response = await instance.get(url);
+    dispatch({
+      type: QUERY_POSSIBLE_ANSWER_CONCEPTS,
+      payload: response.data,
+    });
+  } catch (error) {
+    notify.show('There is something wrong with your connection', 'error', 3000);
+  }
+};
+
+export const addSelectedAnswersToState = answers => (dispatch) => {
+  dispatch({ type: ADD_SELECTED_ANSWERS, payload: answers });
+};
+
+export const changeSelectedAnswer = data => (dispatch) => {
+  dispatch({ type: CHANGE_ANSWER_MAPPING, payload: data });
+};
+
+export const addAnswerMappingToConcept = async (url, source, answers) => {
+  const user = localStorage.getItem('username');
+  const mappingUrl = `/users/${user}/sources/${source}/mappings/`;
+  const data = answers.map(answer => (answer.map_scope === 'Internal' ? {
+    map_type: answer.map_type,
+    from_concept_url: url,
+    to_concept_url: answer.url,
+  }
+    : {
+      map_type: answer.map_type,
+      from_concept_url: url,
+      to_source_url: answer.url,
+      to_concept_code: answer.id,
+      to_concept_name: answer.display_name,
+    }));
+
+  const promises = data.map(dt => instance.post(mappingUrl, dt));
```

Comment:
@AGMETEOR, using a better variable name other than `dt` will be better, you can make it more descriptive.